infrastructures/kubernetes/src/test/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/pvc/CommonPVCStrategyTest.java

```diff
@@ -422,44 +422,6 @@ public void shouldNotDeleteCommonPVCIfUserHasWorkspaces() throws Exception {
     verify(pvcSubPathHelper).removeDirsAsync(WORKSPACE_ID, "ns", PVC_NAME, WORKSPACE_ID);
   }
 
-  @Test
-  public void shouldNotDeleteCommonPVCIfSubjectIsAnonymous() throws Exception {
-    // given
-    Workspace workspace = mock(Workspace.class);
-    Page workspaces = mock(Page.class);
-    KubernetesPersistentVolumeClaims persistentVolumeClaims =
-        mock(KubernetesPersistentVolumeClaims.class);
-
-    when(workspaceManager.getWorkspaces(anyString(), eq(false), anyInt(), anyLong()))
-        .thenReturn((workspaces));
-    when(workspaces.isEmpty()).thenReturn(true);
-    when(workspace.getId()).thenReturn(WORKSPACE_ID);
-
-    WorkspaceConfig workspaceConfig = mock(WorkspaceConfig.class);
-    when(workspace.getConfig()).thenReturn(workspaceConfig);
-
-    Map<String, String> workspaceConfigAttributes = new HashMap<>();
-    when(workspaceConfig.getAttributes()).thenReturn(workspaceConfigAttributes);
-    workspaceConfigAttributes.put(PERSIST_VOLUMES_ATTRIBUTE, "true");
-
-    KubernetesNamespace ns = mock(KubernetesNamespace.class);
-    when(factory.get(eq(workspace))).thenReturn(ns);
-    when(ns.getName()).thenReturn("ns");
-
-    EnvironmentContext context = new EnvironmentContext();
-    context.setSubject(Subject.ANONYMOUS);
-    EnvironmentContext.setCurrent(context);
-
-    // when
-    commonPVCStrategy.cleanup(workspace);
-
-    // then
-    verify(workspaceManager, never()).getWorkspaces(anyString(), eq(false), anyInt(), anyLong());
-    verify(ns, never()).persistentVolumeClaims();
-    verify(persistentVolumeClaims, never()).delete(PVC_NAME);
-    verify(pvcSubPathHelper).removeDirsAsync(WORKSPACE_ID, "ns", PVC_NAME, WORKSPACE_ID);
-  }
```

Comment:
In reference to the if check from https://github.com/eclipse-che/che-server/pull/16#discussion_r642919396, I deleted this test.

It fails without the if check, due to an NPE (workspace is `null`) within the code that was inside the if check's body here: https://github.com/eclipse-che/che-server/blob/f01607926ffbeac3a9978bfd5fcdbd88d5ff4c92/infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/pvc/CommonPVCStrategy.java#L290